pimcore/lib/Pimcore/Console/Command/ThumbnailsImageCommand.php

```diff
@@ -98,7 +98,8 @@ protected function execute(InputInterface $input, OutputInterface $output)
                 }
 
                 if ($input->getOption("system")) {
-                    $thumbnail = Asset\Image\Thumbnail\Config::getPreviewConfig();
+                    $thumbnail = \Pimcore::getDiContainer()->get('Pimcore\Model\Asset\Image\Thumbnail\Config');
+                    $thumbnail = $thumbnail::getPreviewConfig();
```

Comment:
Since it's a static method call there is no instance to create, so it's not even a singleton.

However your suggestion sounds good (putting the whole config in the container), that would really be a singleton now so it makes sense.

> "Pimcore\Model\Asset\Video\Thumbnail\Config::getPreviewConfig" is rather long, but makes sense to use. What would you suggest?

I'm not working with Pimcore at all so I'm not aware of the naming convention. In my projects I would name it something like `pimcore.assets.video.thumbnail.preview-config` or something like that. I tend to use class names for objects (because there is a real class behind), but I use lower-case IDs for stuff like config, etc. Of course that's just my personal usage, not related to Pimcore.